doc/services/smf/index.rst

```diff
@@ -71,6 +78,37 @@ And this example creates three hierarchical states::
    };
 
 
+This example creates three hierarchical states with an initial transition
+from parent state S0 to child state S2::
```

Comment:
This section should be added following the Hierarchical State Machine Example.  It should also include the code showing state transitions.

As is, I don't think this diagram is helpful to the user as it's demonstrating too trivial a case.

But I'm okay with you dropping this example/diagram from this PR and creating a follow-up PR to make the improvements here.

Reply:
I think a good goal here would be to have a diagram that matches the state transitions from the test.

Reply:
Moved diagram to end of document.